modules/spacelift-automation/variables.tf

```diff
@@ -0,0 +1,242 @@
+  type        = bool
+  description = "Flag to enable/disable automatic retry of the stack"
+  default     = false
+}
+
+variable "before_apply" {
+  type        = list(string)
+  description = "List of before-apply scripts"
+  default     = []
+}
+
+variable "before_destroy" {
+  type        = list(string)
+  description = "List of before-destroy scripts"
+  default     = []
+}
+
+variable "before_init" {
+  type        = list(string)
+  description = "List of before-init scripts"
+  default     = []
+}
+
+variable "before_perform" {
+  type        = list(string)
+  description = "List of before-perform scripts"
+  default     = []
+}
+
+variable "before_plan" {
+  type        = list(string)
+  description = "List of before-plan scripts"
+  default     = []
+}
+
+variable "description" {
+  type        = string
+  description = "Description of the stack"
+  default     = "Managed by spacelift-automation Terraform root module."
+}
+
+variable "destructor_enabled" {
+  type        = bool
+  description = "Flag to enable/disable the destructor for the Stack."
+  default     = false
+}
+
+variable "drift_detection_enabled" {
+  type        = bool
+  description = "Flag to enable/disable Drift Detection configuration for a Stack."
+  default     = false
+}
+
+variable "drift_detection_ignore_state" {
+  type        = bool
+  description = <<-EOT
+  Controls whether drift detection should be performed on a stack
+  in any final state instead of just 'Finished'.
+  EOT
+  default     = false
+}
+
+variable "drift_detection_reconcile" {
+  type        = bool
+  description = "Flag to enable/disable automatic reconciliation of drifts."
+  default     = false
+}
+
+variable "drift_detection_schedule" {
+  type        = list(string)
+  description = "The schedule for drift detection."
+  default     = ["0 4 * * *"]
+}
+
+variable "drift_detection_timezone" {
+  type        = string
+  description = "The timezone for drift detection."
+  default     = "UTC"
+}
+
+variable "enable_local_preview" {
+  type        = bool
+  description = "Indicates whether local preview runs can be triggered on this Stack."
+  default     = false
+
+}
+variable "manage_state" {
+  type        = bool
+  description = "Determines if Spacelift should manage state for this stack."
+  default     = false
+}
+
+variable "protect_from_deletion" {
+  type        = bool
+  description = "Protect this stack from accidental deletion. If set, attempts to delete this stack will fail."
+  default     = false
+}
+
+variable "space_id" {
+  type        = string
+  description = "Place the stack in the specified space_id."
+  default     = "root"
+}
+
+variable "terraform_smart_sanitization" {
+  type        = bool
+  description = <<-EOT
+  Indicates whether runs on this will use terraform's sensitive value system to sanitize
+  the outputs of Terraform state and plans in spacelift instead of sanitizing all fields.
+  EOT
+  default     = false
+}
+
+variable "terraform_version" {
```

Comment:
Thoughts on renaming this (and other instances of `terraform_***`) to `tf_version` so less explicit now that we have tofu (and are even defaulting to tofu)?

Reply:
This corresponds to `spacelift_stack` arguments in Spacelift’s provider. I'd prefer to keep them equal. What do you think?